Gems/MiniAudio/Code/CMakeLists.txt

```diff
@@ -0,0 +1,217 @@
+#       in which case it will see if that platform is present here or in the restricted folder.
+#       i.e. It could here in our gem : Gems/MiniAudio/Code/Platform/<platorm_name>  or
+#            <restricted_folder>/<platform_name>/Gems/MiniAudio/Code
+o3de_pal_dir(pal_dir ${CMAKE_CURRENT_LIST_DIR}/Platform/${PAL_PLATFORM_NAME} "${gem_restricted_path}" "${gem_path}" "${gem_parent_relative_path}")
+
+# Now that we have the platform abstraction layer (PAL) folder for this folder, thats where we will find the
+# traits for this platform. Traits for a platform are defines for things like whether or not something in this gem
+# is supported by this platform.
+include(${pal_dir}/PAL_${PAL_PLATFORM_NAME_LOWERCASE}.cmake)
+
+# The MiniAudio.API target declares the common interface that users of this gem should depend on in their targets
+ly_add_target(
+    NAME MiniAudio.API INTERFACE
+    NAMESPACE Gem
+    FILES_CMAKE
+        miniaudio_api_files.cmake
+        ${pal_dir}/miniaudio_api_files.cmake
+    INCLUDE_DIRECTORIES
+        INTERFACE
+            Include
+    BUILD_DEPENDENCIES
+        INTERFACE
+           AZ::AzCore
+)
+
+# The MiniAudio.Private.Object target is an internal target
+# It should not be used outside of this Gems CMakeLists.txt
+ly_add_target(
+    NAME MiniAudio.Private.Object STATIC
+    NAMESPACE Gem
+    FILES_CMAKE
+        miniaudio_private_files.cmake
+        ${pal_dir}/miniaudio_private_files.cmake
+    TARGET_PROPERTIES
+        O3DE_PRIVATE_TARGET TRUE
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            Include
+            Source
+    BUILD_DEPENDENCIES
+        PUBLIC
+            AZ::AzCore
+            AZ::AzFramework
+        PRIVATE
+            3rdParty::miniaudio
+            3rdParty::stb_vorbis
+)
+
+# Here add MiniAudio target, it depends on the Private Object library and Public API interface
+ly_add_target(
+    NAME MiniAudio ${PAL_TRAIT_MONOLITHIC_DRIVEN_MODULE_TYPE}
+    NAMESPACE Gem
+    FILES_CMAKE
+        miniaudio_shared_files.cmake
+        ${pal_dir}/miniaudio_shared_files.cmake
+    INCLUDE_DIRECTORIES
+        PUBLIC
+            Include
+        PRIVATE
+            Source
+    BUILD_DEPENDENCIES
+        PUBLIC
+            Gem::MiniAudio.API
+        PRIVATE
+            Gem::MiniAudio.Private.Object
+)
+
+# By default, we will specify that the above target MiniAudio would be used by
+# Client and Server type targets when this gem is enabled.  If you don't want it
+# active in Clients or Servers by default, delete one of both of the following lines:
+ly_create_alias(NAME MiniAudio.Clients NAMESPACE Gem TARGETS Gem::MiniAudio)
+ly_create_alias(NAME MiniAudio.Servers NAMESPACE Gem TARGETS Gem::MiniAudio)
+
+# For the Client and Server variants of MiniAudio Gem, an alias to the MiniAudio.API target will be made
+ly_create_alias(NAME MiniAudio.Clients.API NAMESPACE Gem TARGETS Gem::MiniAudio.API)
+ly_create_alias(NAME MiniAudio.Servers.API NAMESPACE Gem TARGETS Gem::MiniAudio.API)
```

Comment:
Adding a source file propertie to inject the O3DE_GEM_NAME define into the MiniAudioModule.cpp will complete the work needed to simplify renaming the gem in the future if need be
https://github.com/o3de/o3de/blob/development/Templates/DefaultGem/Template/Code/CMakeLists.txt#L81-L92